### ISSUE TITLE
undefined method `dependency' with Bundler.require

### ISSUE THREAD

**Issue by prettycoder** [lostisland/faraday-excon#4]
Looks like the `dependency` method is missing, maybe removed from faraday ?

Reproduce:

Gemfile
```
source "https://rubygems.org"

gem 'excon'
gem 'faraday'
gem 'faraday-excon'
```

Shell
```
bundle
be irb
```

Irb
```
Bundler.require
/Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/adapter/excon.rb:7:in `<class:Excon>': undefined method `dependency' for Faraday::Adapter::Excon:Class (NoMethodError)
	from /Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/adapter/excon.rb:6:in `<class:Adapter>'
	from /Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/adapter/excon.rb:4:in `<module:Faraday>'
	from /Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/adapter/excon.rb:3:in `<top (required)>'
	from /Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/excon.rb:3:in `require_relative'
	from /Users/maia.engeli/vendor/cache/ruby/3.0.0/gems/faraday-excon-1.1.0/lib/faraday/excon.rb:3:in `<top (required)>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:79:in `require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:79:in `rescue in block in require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:57:in `block in require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:50:in `each'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:50:in `require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler.rb:174:in `require'
	from (irb):1:in `<main>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/bin/irb:23:in `<top (required)>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
	... 14 levels...
/Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:66:in `require': cannot load such file -- faraday-excon (LoadError)
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:66:in `block (2 levels) in require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:61:in `each'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:61:in `block in require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:50:in `each'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/runtime.rb:50:in `require'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler.rb:174:in `require'
	from (irb):1:in `<main>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/bin/irb:23:in `<top (required)>'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `kernel_load'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:28:in `run'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:474:in `exec'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/maia.engeli/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
```

**Comment by olleolleolle** [comment]
The UPGRADING guide in Faraday has a mention, now, of this - the recommendation is "use a regular require, instead of the deferred (and also, hehe, now-removed) "dependency" method.  

Hope this helps!

**Comment by iMacTia** [comment]
Mmmmh although the `dependency` method was removed in Faraday 2.0 🤔
`faraday-excon` has a gemspec dependency to `faraday ~> 1.0`, so how did your bundle load it together with Faraday 2.0?

**Comment by prettycoder** [comment]
@iMacTia 
the gemspec has `add_development_dependency` for `faraday ~> 1.0`, so that version is not loaded by default 🥺
@olleolleolle 
It explains, but that line of code with `dependency` is in this repo. Shall I create a PR?

**Comment by prettycoder** [comment]
no PR needed, it is already gone in main. removed here https://github.com/lostisland/faraday-excon/commit/75a3383b91461a6f7fa8de52abc2602fd4ccd6d0 
Could you release a new gem version?

**Comment by olleolleolle** [comment]
Right, for completeness, the workaround is to use this gem from the GitHub repository:

     gem "faraday-excon", github: "lostisland/faraday-excon"

A new release is needed to make the need for the workaround go away.

**Comment by olleolleolle** [comment]
@prettycoder The new version has been released as 2.0.0. Thanks for reminding!

**Comment by prettycoder** [comment]
thanks for the quick turnaround :)
